superset/views/base_api.py

```diff
@@ -452,15 +483,18 @@ def get_text_for_model(model: Model) -> str:
         # handle filters
         filters = self._get_related_filter(datamodel, column_name, args.get("filter"))
         # Make the query
-        count, values = datamodel.query(
+        _, rows = datamodel.query(
```

Comment:
This is interesting - the endpoint doesn't behave the way I would expect. If I passed multiple filters and one of them was IDs, I would expect it to be a subset of the IDs listed. If I passed only a list of IDs, I would expect only the listed IDs to be returned.  Would it be possible to treat the IDs list as part of the original query? That would also avoid running two queries to build this request.

Reply:
It's possible, but note that passing the filters and the ids the intent is not to have logical AND's on this case, but logical OR's. The goal is to have a generic filterable dropdown list (with pagination) that always includes a certain set of ID's on its response. So we expect a superset ;) of the ID's passed.

Reply:
Then this feels like a bit of a kludge - why is it required? A logical `or` on a filter API that's implicit doesn't make sense. How can we make the `or` explicit in the request from the caller?